src/components/Page/PageTitleBar.vue

```diff
@@ -0,0 +1,343 @@
+		</div>
+
+		<!-- Page title -->
+		<PageTitle v-if="isLandingPage"
+			ref="pageTitle"
+			:value="currentCollective.name"
+			:disabled="true" />
+		<PageTitle v-else-if="isTemplatePage"
+			ref="pageTitle"
+			:value="t('collectives', 'Template')"
+			:disabled="true" />
+		<PageTitle v-else
+			ref="pageTitle"
+			v-model="newTitle"
+			:placeholder="t('collectives', 'Title')"
+			:disabled="!currentCollectiveCanEdit"
+			@blur="onTitleBlur()"
+			@save="$emit('save-editor')"
+			@submit="$emit('focus-editor')" />
+
+		<div class="titlebar-buttons" :class="{'titlebar-buttons_sidebar-toggle': !isMobile && !showing('sidebar')}">
+			<!-- Edit button if editable -->
+			<EditButton v-if="currentCollectiveCanEdit"
+				:mobile="isMobile"
+				class="edit-button" />
+
+			<!-- Actions menu -->
+			<PageActionMenu :show-files-link="!isPublic"
+				:page-id="currentPage.id"
+				:parent-id="currentPage.parentId"
+				:timestamp="currentPage.timestamp"
+				:last-user-id="currentPage.lastUserId"
+				:last-user-display-name="currentPage.lastUserDisplayName"
+				:is-landing-page="isLandingPage"
+				:is-template="isTemplatePage" />
+		</div>
+	</div>
+</template>
+<script>
+import isMobile from '@nextcloud/vue/dist/Mixins/isMobile.js'
+import pageMixin from '../../mixins/pageMixin.js'
+
+import { mapActions, mapState } from 'pinia'
+import { useRootStore } from '../../stores/root.js'
+import { useCollectivesStore } from '../../stores/collectives.js'
+import { usePagesStore } from '../../stores/pages.js'
+import { showError } from '@nextcloud/dialogs'
+
+import { NcButton, NcLoadingIcon } from '@nextcloud/vue'
+import NcEmojiPicker from '@nextcloud/vue/dist/Components/NcEmojiPicker.js'
+import CollectivesIcon from '../Icon/CollectivesIcon.vue'
+import EmoticonOutlineIcon from 'vue-material-design-icons/EmoticonOutline.vue'
+import EditButton from './EditButton.vue'
+import PageActionMenu from './PageActionMenu.vue'
+import PageTemplateIcon from '../Icon/PageTemplateIcon.vue'
+import PageTitle from './PageTitle.vue'
+
+export default {
+	name: 'PageTitleBar',
+
+	components: {
+		CollectivesIcon,
+		EditButton,
+		EmoticonOutlineIcon,
+		NcButton,
+		NcEmojiPicker,
+		NcLoadingIcon,
+		PageActionMenu,
+		PageTemplateIcon,
+		PageTitle,
+	},
+
+	mixins: [
+		isMobile,
+		pageMixin,
+	],
+
+	props: {
+		isFullWidth: {
+			type: Boolean,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			newTitle: '',
+		}
+	},
+
+	computed: {
+		...mapState(useRootStore, [
+			'isPublic',
+			'loading',
+			'showing',
+		]),
+		...mapState(useCollectivesStore, [
+			'currentCollective',
+			'currentCollectiveCanEdit',
+		]),
+		...mapState(usePagesStore, [
+			'currentPage',
+			'isIndexPage',
+			'isTemplatePage',
+			'isLandingPage',
+		]),
+
+		// TODO: remove when we stop supporting NC < 30
+		isPreNc30() {
+			return window.getComputedStyle(document.body).getPropertyValue('--default-clickable-area') === '44px'
+		},
+
+		hasSidebarToggle() {
```

Comment:
@mejo- Seems like `hasSidebarToggle()` is not in-use

Reply:
Good catch, thank you :pray: